build/gulp.migration.js

```diff
@@ -0,0 +1,133 @@
+require('bluebird-global')
+const path = require('path')
+const fs = require('fs')
+const mkdirp = require('mkdirp')
+const exec = require('child_process').exec
+const fse = require('fs-extra')
+const glob = require('glob')
+const tmp = require('tmp')
+const dotenv = require('dotenv')
+
+const dumpServerData = async () => {
+  const args = require('yargs')(process.argv).argv
+
+  const rootPath = args.path || './'
+  const archiveName = args.name || 'test-migration'
+  const pgBinPath = args.pgPath || __dirname
+
+  const tmpDir = tmp.dirSync({ unsafeCleanup: true })
+
+  const dataFolder = await getValidPath(['out/bp/data', 'data'], rootPath)
+  const dataItems = await fse.readdir(dataFolder)
+
+  for (const item of dataItems.filter(x => x !== 'assets')) {
+    await fse.copy(path.resolve(dataFolder, item), `${tmpDir.name}/${item}`)
+  }
+
+  try {
+    mkdirp.sync(`${tmpDir.name}/storage`)
+    await dumpPostgresData(`${tmpDir.name}/storage`, rootPath, pgBinPath)
+  } catch (err) {
+    console.error('Error:', err)
+
+    console.error(`Could not dump the postgres database.
+Postgres must be installed, and the tool pg_dump must be available on your PATH (use --pgPath to override it)
+Please make sure that the credentials provided in DATABASE_URL are valid`)
+    throw err
+  }
+
+  const files = await Promise.fromCallback(cb => glob('**/*', { cwd: tmpDir.name, nodir: true, dot: true }, cb))
+
+  // Not requiring the file at the top of the file, because it may not have been built yet
+  const { createArchive } = require(path.resolve('./out/bp/core/misc/archive'))
+
+  const currentVersion = await getVersion(rootPath)
+  const filename = await createArchive(`./${archiveName}_${currentVersion}.tgz`, tmpDir.name, files)
+
+  console.log(`Complete server archive saved at ${path.join(__dirname, filename)}.`)
+  tmpDir.removeCallback()
+}
+
+const dumpPostgresData = async (storagePath, rootPath, pgBinPath) => {
+  let databaseUrl = process.env.DATABASE_URL
+
+  const dotEnvPath = await getValidPath(['out/bp/.env', '.env'], rootPath)
+  if (dotEnvPath) {
+    const dot = dotenv.config({ path: dotEnvPath })
+
+    if (dot && dot.parsed.DATABASE_URL) {
+      databaseUrl = dot.parsed.DATABASE_URL
+    }
+  }
+
+  if (databaseUrl && databaseUrl.startsWith('postgres')) {
```

Comment:
In this case, I don't really like the false positive when the provided DATABASE_URL isn't a PostgreSQL valid connection URI. I Think it would be better to throw an error so we don't create an archive for nothing.

Reply:
Agreed, but it was just to keep the logic simple for the dump. I'd probably do this differently in the code, but that's just a dev tool, I didn't put much time there :P